project_file_parser.go

```diff
@@ -68,7 +68,7 @@ func (p ProjectFileParser) ParseVersion(path string) (string, error) {
 
 	// This regular expression matches on 'net<x>.<y>',
 	// 'net<x>.<y>-<platform>' & 'netcoreapp<x>.<y>'
-	targetFrameworkRe := regexp.MustCompile(`net(?:coreapp)?(\d\.\d)(?:\w+)?`)
+	targetFrameworkRe := regexp.MustCompile(`net(?:coreapp)?(?:(\d\.\d)(?:\-?\w+)?)$`)
 	for _, group := range project.PropertyGroups {
 		matches := targetFrameworkRe.FindStringSubmatch(group.TargetFramework)
 		if len(matches) == 2 {
```

Comment:
Taking a look at this loop that iterates over the property groups in the XML file, it seems like the error we currently return is inaccurate/misleading. We don't directly check for the presence of the `<TargetFramework>` tag. Maybe rewording the error to say something like "missing _or invalid_ TargetFramework property" would help clarify things.